engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/WorldSetupScreen.java

```diff
@@ -54,82 +51,42 @@ public class WorldSetupScreen extends CoreScreenLayer {
-            } else if (customWorldName.getText().equalsIgnoreCase(world.getWorldName().toString())) {
-                //same name as before: go back to universe setup
-                goBack = true;
-            } else {
-                //no match found: go back to universe setup
-                goBack = true;
-            }
+        WidgetUtil.trySubscribe(this, "close", button -> {
 
-            if (goBack) {
-                newWorldName = new Name(customWorldName.getText());
-                world.setWorldName(newWorldName);
-                triggerBackAnimation();
-            }
+            triggerBackAnimation();
         });
     }
-
-    /**
-     * This method sets the world name in title as well as in UITextBox
-     *
-     * @param customWorldName
-     */
-    private void setCustomWorldName(UIText customWorldName) {
-        customWorldName.setText(world.getWorldName().toString());
-    }
-
     @Override
     public void onOpened() {
         super.onOpened();
 
-        UILabel subitle = find("subtitle", UILabel.class);
-        subitle.setText(translationSystem.translate("${engine:menu#world-setup}") + " for " + world.getWorldName().toString());
-        UIText customWorldName = find("customisedWorldName", UIText.class);
-        setCustomWorldName(customWorldName);
+        UILabel subtitle = find("subtitle", UILabel.class);
+        subtitle.setText(translationSystem.translate("${engine:menu#world-setup}") + " for " + universe.getGameName().toString());
     }
 
     /**
      * This method sets the world whose properties are to be changed. This function is called before the screen comes
      * to the forefront.
      *
      * @param subContext    the new environment created in {@link UniverseSetupScreen}
-     * @param worldSelected the world whose configurations are to be changed.
+     * @param universe      the universe whose world's configurations are to be changed.
      * @throws UnresolvedWorldGeneratorException
      */
-    public void setWorld(Context subContext, WorldSetupWrapper worldSelected)
-            throws UnresolvedWorldGeneratorException {
-        world = worldSelected;
+    public void setWorld(Context subContext, UniverseWrapper universe) {
+        this.universe = universe;
         context = subContext;
-        SimpleUri worldGenUri = worldSelected.getWorldGeneratorInfo().getUri();
         environment = context.get(ModuleEnvironment.class);
         context.put(WorldGeneratorPluginLibrary.class, new TempWorldGeneratorPluginLibrary(environment, context));
-        if (world.getWorldGenerator() == null) {
-            worldGenerator = WorldGeneratorManager.createWorldGenerator(worldGenUri, context, environment);
-            world.setWorldGenerator(worldGenerator);
-        } else {
-            worldGenerator = world.getWorldGenerator();
-        }
+        worldGenerator = universe.getWorldGenerator();
```

Comment:
The world generator in the universe can no longer be `null`?

Reply:
world generator is set at initialize time (see UniverseSetupScreen#168)
I'd say it not being set is sufficiently unlikely that we don't need the null check anymore.
if you disagree I can re-add it of course